### ISSUE TITLE
env deploy fails when adding certificates to manifest.yml

### ISSUE THREAD

**Issue by srbartlett** [aws/copilot-cli#5585]
Hi and happy new year.

I am using copilot version: v1.32.0

I would like to add certifications to my environment manifest.  I have manually created the Certification in ACM and updated the public certificates section as described in [manifest environment](https://aws.github.io/copilot-cli/docs/manifest/environment/)

Running: `copilot env deploy -n XXX` results in the following error message

```
 Export [APP_NAME]-[ENV_NAME]-SubDomain cannot be deleted as it is in use by
   [APP_NAME]-[ENV_NAME]
```

When I inspect the cloudformation template as output by `copilot env package` I can confirm the following section is **not** present in the `outputs` section when `certificates` are defined in the manifest. 

```
  EnvironmentSubdomain:
    Condition: DelegateDNS
    Value: !Sub ${EnvironmentName}.${AppName}.${AppDNSName}
    Description: The domain name of this environment.
    Export:
      Name: !Sub ${AWS::StackName}-SubDomain
```

For some reason the existence of certificates is deleting the environment subdomain.   

The application has been upgraded using `copilot app upgrade`.

The only curve ball is the environment I am attempting to deploy to is in a different region.

Any help would be appreciated.  Thanks

**Comment by KollaAdithya** [comment]
Hello @srbartlett !

Happy new year!

1. Do you have any Copilot service especially a LBWS deployed in that environment ?  

I would assume yes for the above question, there should be a stack that is importing `EnvironmentSubdomain`.

you can list all the stacks that import an exported output value using below command 
`aws cloudformation list-imports --export-name <appName>-<envName>-SubDomain`  

To get around this there are two ways.

**Option1**
 Delete all the service stacks that are importing `EnvironmentSubdomain` and then use `http.public.certificates` to import certificates.

**Option2** 
you can use yaml patches to add back that output .

Run the below commands 
1. copilot env override —tool yamlpatch  
```
- op: add
  path: /Outputs/EnvironmentSubdomain
  value:
    Condition: DelegateDNS
    Value: !Sub ${EnvironmentName}.${AppName}.${AppDNSName}
    Description: The domain name of this environment.
    Export:
      Name: !Sub ${AWS::StackName}-SubDomain
  ```
2. Redeploy the environment with `http.public.certificates` using `copilot env deploy`.

**Comment by srbartlett** [comment]
Hi @KollaAdithya 

Thanks for your suggestions -- very helpful and informative.  I wasn't aware of the `env override` command so thanks for sharing that. 

I went with **Option 1** as the environment is new thus easy to destroy.  After redeploying, the Certificate was successfully associated to the ALB as planned.  I had to add the `http: alias:` to the LBSW manifest which makes sense.  I was expecting a HostedZone to be created but I guess it needs to be manual with this approach? 
 
Thanks again 😄
